python/simulation.py

```diff
@@ -1812,22 +1834,34 @@ def load_minus_force_data(self, force, fdata):
         self.load_force_data(force, fdata)
         force.scale_dfts(complex(-1.0))
 
-    def add_flux(self, fcen, df, nfreq, *fluxes):
-        flux = DftFlux(self._add_flux, [fcen, df, nfreq, fluxes])
+    # TODO: make add_flux_uneven, wrap add_flux around it, change parameters of _add_flux
+    def add_flux_uneven(self, freqs, nfreq, *fluxes):
+        freqs.sort()
+        flux = DftFlux(self._add_flux, [freqs, nfreq, fluxes])
         self.dft_objects.append(flux)
         return flux
 
-    def _add_flux(self, fcen, df, nfreq, fluxes):
+    def add_flux(self, fcen, df, nfreq, *fluxes):
+        freqs = np.linspace(fcen - df / 2, fcen + df / 2, nfreq).tolist()
```

Comment:
You can remove the `tolist()` call since the typemaps are expecting a numpy array.